tests/testsuite.src/syn_misc.at

```diff
@@ -8592,3 +8592,93 @@ prog.cob:9: warning: DISPLAY statement in DEFAULT SECTION is not implemented
 AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [OK], [])
 
 AT_CLEANUP
+
+
+AT_SETUP([PICTURE L])
+AT_KEYWORDS([extensions gcos picture-l])
+
+AT_DATA([prog_extraneous_depending.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_extraneous_depending.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 L		COMP-1.
+       1 F-DAT		PICTURE X(10) DEPENDING ON L.
+       PROCEDURE        DIVISION.
+           MOVE SPACES TO F-DAT
+           STOP RUN.
+])
+AT_CHECK([$COMPILE_ONLY prog_extraneous_depending.cob], [1], [],
+[prog_extraneous_depending.cob:7: error: DEPENDING clause must either come with an OCCURS clause or a PICTURE string with 'L' character
+])
+
+AT_DATA([prog_missing_depending.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_missing_depending.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 F-DAT		PICTURE LX(10).
+       PROCEDURE        DIVISION.
+           MOVE SPACES TO F-DAT
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE_ONLY prog_missing_depending.cob], [1], [],
+[prog_missing_depending.cob:6: error: missing DEPENDING clause for PICTURE string with 'L' character
+])
+
+AT_DATA([prog_value.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_value.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 L		COMP-1 VALUE 10.
+       1 F-DAT		PICTURE LX(10) DEPENDING ON L
+                        VALUE "0123456789".
+       PROCEDURE        DIVISION.
+           STOP RUN.
+])
+AT_CHECK([$COMPILE_ONLY prog_value.cob], [1], [],
+[prog_value.cob:7: error: PICTURE L and VALUE are mutually exclusive
```

Comment:
I suggest to only reference `L` during parsing of the `PICTURE` string. Let's stay with "variable-length PICTURE" - in this case as `_("variable-length PICTURE")` in all other places (I may have not spot all of those yet).